app/scripts/views/settings/account_recovery/confirm_revoke.js

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Cocktail = require('cocktail');
+const FormView = require('../../form');
+const PasswordMixin = require('../../mixins/password-mixin');
+const ModalSettingsPanelMixin = require('../../mixins/modal-settings-panel-mixin');
+const Template = require('templates/settings/account_recovery/confirm_revoke.mustache');
+
+const View = FormView.extend({
+  template: Template,
+  className: 'account-recovery-confirm-revoke',
+  viewName: 'settings.account-recovery.confirm-revoke',
+
+  events: {
```

Comment:
You can access this page w/o having set a recovery key just by browing to `/settings/account_recovery/confirm_revoke`. Maybe in a `beforeRender` or `afterRender`, ensure the user *has* a recovery key or else send them to the setup page.

Reply:
Added!